mirascope/wandb.py

```diff
@@ -162,21 +219,24 @@ def call_with_trace(
         """
         try:
             start_time = datetime.datetime.now().timestamp() * 1000
-            response = super().call()
-            span = trace(self, response, parent, **kwargs)
-            return response, span
+            response = self.call(**kwargs)
+            tool_type = None
+            if response.tool_types and len(response.tool_types) > 0:
+                tool_type = response.tool_types[0].__bases__[0]  # type: ignore
+            span = trace(self, response, tool_type, parent, **kwargs)
+            return response, span  # type: ignore
         except Exception as e:
             return None, trace_error(self, e, parent, start_time, **kwargs)
```

Comment:
I know this was like this b4 but, is there a reason we dont raise?

Reply:
Because we're catching and handling the error here. So to check for error you check if response is None (and the error is traced to wandb)